code/modules/mob/living/carbon/human/life.dm

```diff
@@ -1802,7 +1803,15 @@
 	if(thermal_protection == 1) // Immune.
 		return
 	else
-		bodytemperature += (BODYTEMP_HEATING_MAX + (fire_stacks * 15)) * (1-thermal_protection)
+		var/fire_temp_add = (BODYTEMP_HEATING_MAX + (fire_stacks * 15)) * (1-thermal_protection)
+		//This is to prevent humans from heating up indefinitely. A human being on fire (fat burns at 250C) can't magically
+		//   increase your body temperature beyond 250C, but it's possible something else (atmos) has heated us up beyond it,
+		//   so don't worry about the firestacks at that point. Really, we should be cooling the room down, because it has
+		//   to expend energy to heat our body up! But let's not worry about that.
+		if((bodytemperature + fire_temp_add) > HUMAN_COMBUSTION_TEMP)
+			return
+
+		bodytemperature += fire_temp_add

```

Comment:
Should this not just set it to `HUMAN_COMBUSTION_TEMP` rather than prevent a temperature increase altogether?

Reply:
i.e., `bodytemperature = min(bodytemperature + fire_temp_add, HUMAN_COMBUSTION_TEMP)`

Reply:
This suggestion doesn't make any sense. It means that if someone has been heated up by a room to 2000k and then catches on fire, their body temperature will lower to ~500k.